components/camel-file-watch/src/main/java/org/apache/camel/component/file/watch/FileWatchComponent.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.file.watch;
+
+import java.util.Map;
+
+import io.methvin.watcher.hashing.FileHasher;
+import org.apache.camel.Endpoint;
+import org.apache.camel.spi.Registry;
+import org.apache.camel.spi.annotations.Component;
+import org.apache.camel.support.DefaultComponent;
+
+/**
+ * Represents the component that manages {@link FileWatchEndpoint}.
+ */
+@Component("file-watch")
+public class FileWatchComponent extends DefaultComponent {
+    public static final String EVENT_TYPE_HEADER = "CamelFileEventType";
+
+    protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
+        Endpoint endpoint = new FileWatchEndpoint(uri, remaining, this);
+        setProperties(endpoint, parameters);
+        return endpoint;
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+        Registry registry = getCamelContext().getRegistry();
+        if (registry.lookupByNameAndType("murmur3FFileHasher", FileHasher.class) == null) {
```

Comment:
These bean names, are the from that 3rd party library? Can you add some code comments why you do this

Reply:
These beans are not required for the component. I have added these two into registry, to make it easier to use for users (It is defined as lambda constant in FileHasher class and users could find it hard to register in spring as it would require FieldRetrievingFactoryBean). 

In the last commit it is moved to `doInit`, so the `if` check is not needed. Also added some comments on the method.  I have not found in code any similiar approach in components, so I`m not against removing this registration, if it is bad practise. I can instead add some examples to docs showing how to register `lastModifiedTimeFileHasher` in java-dsl and spring-dsl.